tst/functions/assertions/HaveParameter.Tests.ps1

```diff
@@ -345,6 +345,46 @@ InPesterModuleScope {
             $err.Exception.Message | Verify-Equal "Expected command Invoke-DummyFunction to have a parameter ParamWithNotNullOrEmptyValidation, which is mandatory, of type [System.TimeSpan] and the default value to be 'wrong value', because of reasons, but it wasn't mandatory, it was of type [System.DateTime] and the default value was '(Get-Date)'."
         }
 
+        It 'passes when object parameter has default parameter value $null' {
+            function Test-Parameter {
+                param ( [Parameter()] [object] $objParam = $null )
+            }
+
+            Get-Command Test-Parameter | Should -HaveParameter 'objParam' -Type 'object' -DefaultValue $null
+        }
+
+        It 'fails when object parameter expects default parameter value $null, but there is none' {
+            function Test-Parameter {
+                param ( [Parameter()] [object] $objParam )
+            }
+
+            { Get-Command Test-Parameter | Should -HaveParameter 'objParam' -Type 'object' -DefaultValue $null } | Should -Throw "aaaa"
+        }
+
+        It 'passes when integer parameter has default parameter value 0' {
+            function Test-Parameter {
+                param ( [Parameter()] [int] $intParam = 0 )
+            }
+
+            Get-Command Test-Parameter | Should -HaveParameter 'intParam' -Type 'int' -DefaultValue 0
+        }
+
+        It 'passes when bool parameter has default parameter value $true' {
+            function Test-Parameter {
+                param ( [Parameter()] [bool] $boolParam = $true )
```

Comment:
This test fails, because the param has value, but we don't lift the value from it properly because there is none, because $true is variable descriptor.